internal/core/project.go

```diff
@@ -124,7 +128,11 @@ func NewProject(ctx context.Context, os ...Option) (*Project, error) {
 
 	// Initialize all the applications and load all their components.
 	for _, name := range opts.Config.Apps() {
-		appConfig, err := opts.Config.App(name, nil)
+		// Set input variables for applications and components
+		evalCtx := config.EvalContext(nil, p.dir.DataDir()).NewChild()
+		config.AddVariables(evalCtx, &p.variables)
```

Comment:
It's a little odd to use pointers to maps, and since you just deref the pointer in `AddVariables` I'd probably just change `AddVariables` to take the map without a pointer.

Reply:
Done!